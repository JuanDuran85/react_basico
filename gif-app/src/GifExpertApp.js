import React, { useState } from 'react';
// import PropTypes from 'prop-types';

const GifExpertApp = () => {
    
    //const categorias = ['Risas', 'Homero', 'No'];

    const [categories, setCategories] = useState(['Risas', 'Homero', 'No']);

    const agregarValue = () => {
        // setCategories([...categories, 'nuevo elemento']);
        setCategories(categories => [...categories, 'nuevo Elemento']);
    }

    return (
        <div>
            <h1>Gif Experct App</h1>
            <hr />

            <button onClick={agregarValue}>Agregar</button>

            <ol>
                {
                    categories.map(category => {
                        return <li key={category}>{category}</li>
                    })
                }
            </ol>
        </div>
    )
}

// GifExpertApp.propTypes = {}

export default GifExpertApp
