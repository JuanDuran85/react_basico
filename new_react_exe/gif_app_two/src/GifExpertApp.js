import React, { useState } from 'react';
import { AddCategories } from './components/AddCategories';
import { GifGrid } from './components/GifGrid';

export const Gifexpertapp = () => {

    const [categories, setcategories] = useState(['Homero']);

    return (
        <>
            <h1>GifExpertApp</h1>
            <AddCategories setCategories = { setcategories }/>
            <hr />
            
            <div>
                {
                    categories.map(category => <GifGrid key = { category } categoria = {category} />)
                }
            </div>
        </>
    );
}
