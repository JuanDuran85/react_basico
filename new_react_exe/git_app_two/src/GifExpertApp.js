import React, { useState } from 'react';

export const Gifexpertapp = () => {

    const [categories, setcategories] = useState(['Homero','Laugh','Wtf', 'What']);

    const handleAdd = () => {
        setcategories([...categories, 'Nueva categoria']);
    }

    return (
        <>
            <h1>GifExpertApp</h1>
            <hr />
            <button onClick={handleAdd}>Agregar</button>
            <ol>
                {
                    categories.map(category => <li key={category}>{category}</li>)
                }
            </ol>
        </>
    );
}
