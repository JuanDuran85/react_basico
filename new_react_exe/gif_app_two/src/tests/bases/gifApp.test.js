import { Gifexpertapp } from '../../GifExpertApp';
import { shallow } from 'enzyme';

import "../setupTests";

describe('Test One Componente', () => {
    
    let wrapper = shallow(<Gifexpertapp />);

    beforeEach(() => {
        wrapper = shallow(<Gifexpertapp />);
    })
    
    test('Debe mostrar el componente correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });
    
});
