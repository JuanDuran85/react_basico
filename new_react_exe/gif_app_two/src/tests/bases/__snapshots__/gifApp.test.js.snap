// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test One Componente Debe mostrar el componente correctamente 1`] = `
<Fragment>
  <h1>
    GifExpertApp
  </h1>
  <AddCategories
    setCategories={[Function]}
  />
  <hr />
  <div>
    <GifGrid
      categoria="Homero"
      key="Homero"
    />
  </div>
</Fragment>
`;
